File: user/event/publish.py
import os

from rabbit_helper.rabbit import Rabbit


async def publish_message(routing_key: str, data: dict) -> bool:
    rabbitmq_url = f"amqp://{os.getenv('RABBITMQ_USERNAME')}:{os.getenv('RABBITMQ_PASSWORD')}@{os.getenv('RABBITMQ_HOST')}:{os.getenv('RABBITMQ_PORT')}/"
    rabbit = Rabbit(rabbitmq_url)

    try:
        ok = await rabbit.publish(routing_key, data)
        return ok
    finally:
        rabbit.close()

File: mealPlanning/event/publish.go
package event

import (
	"fmt"
	"log"
	"os"

	Rabbit "github.com/arshia-rgh/rabbit-helper-go/rabbit_helper"
)

func Publish(routingKey string, data any) {
	rabbitMQURL := fmt.Sprintf("amqp://%v:%v@%v:%v/",
		os.Getenv("RABBITMQ_USERNAME"),
		os.Getenv("RABBITMQ_PASSWORD"),
		os.Getenv("RABBITMQ_HOST"),
		os.Getenv("RABBITMQ_PORT"),
	)

	rabbit := Rabbit.New(rabbitMQURL)
	defer rabbit.Close()

	go func() {
		err := rabbit.Publish(routingKey, data)
		if err != nil {
			log.Printf("error publishing mssage to the %v queue, message: %v", routingKey, data)

		}
	}()

}
